components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineSession.kt

```diff
@@ -338,13 +348,18 @@ class SystemEngineSession(
 
     /**
      * See [EngineSession.toggleDesktopMode]
+     *
+     * Precondition:
+     * If settings.useWideViewPort = true, then desktopMode is always on (will never be toggled)
+     * If settings.useWideViewPort = false or null, then desktopMode can be toggled on/off
      */
     override fun toggleDesktopMode(enable: Boolean, reload: Boolean) {
         val webSettings = webView.settings
-        webSettings.userAgentString = toggleDesktopUA(webSettings.userAgentString, enable)
-        webSettings.useWideViewPort = enable
+        val toggle = if (settings.useWideViewPort == true) true else enable
+        webSettings.userAgentString = toggleDesktopUA(webSettings.userAgentString, toggle)
+        webSettings.useWideViewPort = toggle
 
-        notifyObservers { onDesktopModeChange(enable) }
+        notifyObservers { onDesktopModeChange(toggle) }
```

Comment:
Here we still want to pass in `enable` because we enabled desktop mode regardless of `useWideViewPort` (we do more than that, e.g. right now changing the UA).